core/templates/dev/head/exploration_editor/ExplorationEditor.js

```diff
@@ -624,40 +628,145 @@ oppia.controller('ExplorationSaveAndPublishButtons', [
         explorationData.explorationId);
       alertsService.clearWarnings();
 
-      // If the objective has not yet been specified, open the pre-publication
+      var additionalMetadataNeeded = (
+        !explorationTitleService.savedMemento ||
+        !explorationObjectiveService.savedMemento ||
+        !explorationCategoryService.savedMemento ||
+        !explorationLanguageCodeService.savedMemento);
+
+      // If the metadata has not yet been specified, open the pre-publication
       // 'add exploration metadata' modal.
-      if (!explorationObjectiveService.savedMemento) {
+      if (additionalMetadataNeeded) {
         $modal.open({
-          templateUrl: 'modals/addExplorationMetadata',
-          backdrop: true,
-          controller: [
-            '$scope', '$modalInstance', 'explorationObjectiveService',
-            function($scope, $modalInstance, explorationObjectiveService) {
-              // If no objective has been specified yet, require the creator
-              // to specify it.
-              $scope.explorationObjectiveService = explorationObjectiveService;
-              $scope.requireObjectiveToBeSpecified = (
-                !explorationObjectiveService.savedMemento);
-
-              $scope.save = function() {
-                if ($scope.requireObjectiveToBeSpecified) {
-                  if (!explorationObjectiveService.displayed) {
-                    throw Error('Please specify an objective');
-                  }
-                  explorationObjectiveService.saveDisplayedValue();
-                }
-                $modalInstance.close();
-              };
-
-              $scope.cancel = function() {
-                $modalInstance.dismiss('cancel');
-                alertsService.clearWarnings();
-              };
-            }
-          ]
-        }).result.then(function() {
-          saveDraftToBackend('Add an objective.', openPublishExplorationModal);
-        });
+           templateUrl: 'modals/addExplorationMetadata',
```

Comment:
So -- the Travis check is complaining that this should be indented by 10 spaces. This line, however, is indented by 11 spaces.

I suggest removing one space :-) Ditto elsewhere.